src/tutorials/0008-introduction-to-libp2p/04.md

```diff
@@ -0,0 +1,21 @@
+---
+    title: "libp2p in the OSI Model"
+    type: "multiple-choice"
+---
+
+So where does libp2p fit into the current web?
+Usually, we look at the **OSI Model (Open Systems Interconnection Model)** to understand which layer a protocol tries to tackle.
+
+<div class="flex justify-center mv4">
+    <img src="/tutorial-assets/T0008L04-osi-model.png">
+</div>
+
+But the truth is the `OSI Model` as well as the `TCP/IP Model` are just conceptual models. The actual implementations in the current web do not follow these models exactly, but instead it's a mix, i.e., the `TCP/IP Protocol Suite`.
+
+In the end, these models have some shortcomings:
+
+- Actions are repeated through multiple layers (repeated work) because data can go through multiplexing multiple times, or authentication multiple times or even discovery multiple times - it's very inefficient.
+- Information is hidden between layers missing significant improvement opportunities and some protocols have been created with these opportunities in mind by trying to put together as many layers as possible into a single protocol module (an example of this is the [QUIC](https://www.chromium.org/quic) protocol, a multiplexed stream transport over `UDP`).
+
+`libp2p` set out to do something different when it became apparent that following these rigid conceptual models was not ideal - others were already breaking them.
+So by allowing users to pick the pieces of a network stack they need and bundle them together, `libp2p` is creating a new way to maintain and upgrade applications with a secure, reliable and modular network stack.
```

Comment:
I think an accessible metaphor for this pick and choose thing would be helpful. What real world concepts for less techy people can we compare it to?

Reply:
Proposed metaphor:

> We can compare it to buying a couch. Sure, you can buy one after long weeks of searching for the perfect one that you like, but in the end, it's still not exactly as you wanted it to be.
But instead, you can go to an online store that allows you to order your own couch setup with exactly the colours you want and the materials you want.
This couch company as acknowledged the basic building blocks that make up a couch (1-seater, 2-seater or 3-seater? with or without chaise lounge? Velvet, fiber or leather?), and it has provided a web service that allows customers to build the couch they want.
This is the modularity that `libp2p` provides. It has formalised the building blocks into interfaces (Transport, Discovery, etc) so that they can be implemented in different ways (TCP, UDP, etc).

Reply:
This metaphor is super helpful. Might consider moving earlier in the tutorial to hammer on modularity before getting into the specific options libp2p users choose between.

Reply:
Added here: https://github.com/ProtoSchool/protoschool.github.io/pull/574/files#diff-d743b1bc993dad59812891a94c60107b24273f079e6910b9f3c9b3b4b8b2dcc9R84